app/questionnaire/rules.py

```diff
@@ -16,30 +16,11 @@ def evaluate_rule(when, answer_value):
     match_value = when['value'] if 'value' in when else None
     condition = when['condition']
 
-    result = False
-
     # Evaluate the condition on the routing rule
-    if condition == 'equals' and match_value == answer_value:
-        result = True
-    elif condition == 'not equals' and match_value != answer_value:
-        result = True
-    elif condition == 'contains' and isinstance(answer_value, list) and match_value in answer_value:
-        result = True
-    elif condition == 'not contains' and isinstance(answer_value, list) and match_value not in answer_value:
-        result = True
-    elif condition == 'not set':
-        result = answer_value is None
-    elif condition == 'greater than' and isinstance(answer_value, int) and answer_value > match_value:
-        result = True
-    elif condition == 'less than' and isinstance(answer_value, int) and answer_value < match_value:
-        result = True
-
-    return result
+    return evaluate_condition(condition, match_value, answer_value)
```

Comment:
👍 A Positive change

Reply:
I should have done it with the date routing stuff really so better late than never.